actionpack/lib/action_dispatch/routing/url_for.rb

```diff
@@ -180,7 +180,8 @@ def url_for(options = nil)
         when Symbol
           HelperMethodBuilder.url.handle_string_call self, options
         when Array
-          polymorphic_url(options, options.extract_options!)
+          components = options.dup
+          polymorphic_url(components, components.extract_options!)
```

Comment:
`polymorphic_url` doesn't modify the object;
the `url_for` methods in actionpack and actionview
use `options.extract_options!` to split out the options passed to it.

Or do you mean we could update `polymorphic_url`
to accept `[:a, :b, c: true]` directly,
and handle the dup / extract there instead?